js/services/http.js

```diff
@@ -26,7 +27,7 @@ define(function(require, exports) {
       if (payload && payload instanceof FormData) {
         params.body = payload;
       } else if (payload) {
-        params.body = JSON.stringify(payload);
+        params.body = ko.toJSON(payload);
```

Comment:
So, this is an interesting change:  one one hand, I'd argue that the http service does not need to have a dependency on knockout, and it should be expected that the callers passing in their payload perform the appropriate ko.toJS() on their object before sending it over to the http service.

Reply:
On the other hand, it's much better to have one single point of truth for data preparing

Reply:
Thanks @johnSamilin - I had actually made a fix in the way that @chrisknoll described whereby I used ko.toJS() further upstream which fixed this. I'm concerned that changing this in the http service might have other unintended consequences?

Reply:
Great - sounds good in terms of compatibility. To @chrisknoll's other point: not sure if knockout is already a dependency in the @ohdsi/ui-toolbox in which case using it in the http service works. However, if you are establishing the knockout dependency to address this problem, I'd suggest we instead fix this upstream in the Atlas code. When I used ko.toJS() to unwrap the observable before posting to the REST endpoint that fixed the problem without having to touch the existing http service.

Reply:
Hi, the ko.toJS (not toJSON) is to strip the observables away and populate just the raw values.  This isn't about strigify.  What you need to do, when submitting the code over HTTP, is to stringify.  All I was suggesting is that pepole should know if they use HTTP service, they should take care of de-observable their models and the HTTP service should not take any additional steps on behalf of the caller.  This makes it more generic, and less dependency laiden.

Reply:
Agreed @chrisknoll - if the http service sticks with the original approach using JSON.stringify that works for me. In the specific case of the resolveConceptSetExpression function, adding the ko.toJS() was necessary to populate the raw values prior to POST-ing the data via the http service which solved the problem I reported last night.

Reply:
But ko.toJSON does just the same. Doing so in httpService is much easier and will lead us to less boilerplate. By the way it will reduce the chance of a wrong behaviour in the future beacuse it's dramatically easy to forget to call ko.toJS before sending the data to httpService

Reply:
That's true, @johnSamilin , that there's no harm in calling  ```ko.toJSON``` even on plain old javascript objects.  And it will keep people from making the mistake of not calling ko.toJS() before calling a http service method.

All I'm saying is:  1) if you do it your way, then if you wanted to export the HTTP module and use it in another app. you've made the other app dependent on KO.  What if you wanted to make this a stand-alone module for a React app? Wouldn't it be strange to have a KO dependency in a React app?   And 2) What if there are other frameworks at play that need special handling for their javascript objects prior to being sent over HTTP? Do we need to bundle all possible use-cases to handle the proper javascript structure in the HTTP service, or put it on the plate of the person calling the HTTP method?

The pro of your approach is it catches the specfic knockout usecase error. The con of it is that it may not be completely comprehensive (point 2 above) and it introduces a dependency that may add a burdeon of re-usability in different applications if we ever want to pull the HTTP service modules out into a shareable library.

Because of the tight deadlines we're facing right now, I don't think it's worth spending any more time changing the code. But, we should keep these points in mind in the future if either point 1 or 2 I made above become a real issue.

Reply:
@chrisknoll , we already have the http service as a standalone module (get a look at `const OHDSIApi = require('ohdsi-api').Api;`) which is used in Athena (React app). And the module obviously doesn't depend on Knockout. What you see here is child class which extends the module and is required to handle specifics of Atlas (so I'd call it "adapter" in terms of patterns). And as for me, it looks very reasonable to have application-specific `ko.toJSON` here. So I am confused where is the problem

Reply:
Thanks @pavgra : the piece i was missing is that the Http serivce in atlas was an extention to a shared library that was created specifically for atlas.  IN that case, my points about depenency and reusability are moot.

Thanks for the clarification.